cesium/internal/domain/db.go

```diff
@@ -138,6 +140,183 @@ func (db *DB) NewIterator(cfg IteratorConfig) *Iterator {
 	return i
 }
 
+func (db *DB) GetBounds() (tr telem.TimeRange) {
+	db.idx.mu.RLock()
+	defer db.idx.mu.RUnlock()
+	p := db.idx.mu.pointers[0]
+	tr.Start = p.Start
+	p = db.idx.mu.pointers[len(db.idx.mu.pointers)-1]
+	tr.End = p.End
+
+	return tr
+}
+
+// Note that tr is ONLY used to set the tombstone information and is NOT used to find the pointers needed
+func (db *DB) Delete(ctx context.Context, startPosition int, endPosition int, startOffset int64, endOffset int64, tr telem.TimeRange) error {
+	db.idx.mu.RLock()
+	start, ok := db.idx.get(startPosition)
+	if !ok {
+		db.idx.mu.RUnlock()
+		return errors.New("Invalid starting position")
+	}
+	end, ok := db.idx.get(endPosition)
+	if !ok {
+		db.idx.mu.RUnlock()
+		return errors.New("Invalid ending position")
+	}
+	db.idx.mu.RUnlock()
+
```

Comment:
So this is what I think the process youre doing in this function:
1. you acquire a read lock to get the start and end positions
2. you release the read lock and are holding no locks 
3. you acquire a write lock and then do more work involving the start and end positions

Can you guarantee that in step 2, another thread (goroutine?) doesn't come in and alter the start and end positions? I am not convinced that you can.